.github/workflows/update-manpage.yml

```diff
@@ -67,11 +67,9 @@ jobs:
 
       - name: Push commits
         if: steps.update.outputs.committed == 'true'
-        uses: Homebrew/actions/git-try-push@master
-        with:
-          token: ${{ secrets.HOMEBREW_GITHUB_API_TOKEN }}
-          branch: ${{ steps.update.outputs.branch }}
-          force: true
+        run: |
+          git remote set-url origin https://x-access-token:${{ secrets.HOMEBREW_GITHUB_API_TOKEN }}@github.com/${{ github.repository }}
+          git push --force-with-lease origin ${{ steps.update.outputs.branch }}
```

Comment:
This will be much less robust. Can we try to fix up `git-try-push`?

Reply:
Sure, I could open a PR in `Homebrew/actions` to add a flag that does `push --force-with-lease` on the first try too.

Reply:
I'm somewhat mixed on `git-try-push` here. I understand its purpose is to rebase if a push fails, and it does a good job at that, but I don't think the `force` functionality of it makes much sense.

If we care about new commits, the way we are trying to resolve `--force-with-lease` failures automatically conflicts with our intentions. We end up trying to rebase, which is producing undesirable results - as what we actually want is a clean branch off the latest master discarding all old commits made by the workflow.

If we actually don't care about new commits between the workflow starting and the push happening, then can't we just `git push --force` without any retry logic?

Reply:
> If we actually don't care about new commits between the workflow starting and the push happening, then can't we just `git push --force` without any retry logic?

Yes, that makes sense where you want to always just overwrite.

Reply:
Based on Bo's comments, should I modify the `force` functionality to just force-push without retries or the rebase logic (in Homebrew/actions#151)? Or should I be changing just this workflow to `git push --force`?

Reply:
We have other workflows that could do with similar behaviour here so I think it'd be good to have a new/separate mode for `git-try-push` that we use for pushing to non-`master`/PR branches and overwrite everything but that will still attempt to retry. I still think `--force-with-lease` is always a better idea than `--force`.

Reply:
> overwrite everything but that will still attempt to retry

And what should it do when retrying? `git pull --rebase` is not what we want in this case.

Reply:
Rebase on `origin/master` rather than the PR branch.